tests/test_perf.py

```diff
@@ -49,3 +54,97 @@ def test_perf(
         assert is_function_in_collapsed(";_start;__libc_start_main;main;", process_collapsed) ^ bool(
             perf_mode not in ("dwarf", "smart")
         )
+
+
+@pytest.mark.parametrize("runtime", ["native_change_comm"])
+@pytest.mark.parametrize("perf_mode", ["fp"])  # only fp is enough
+@pytest.mark.parametrize("in_container", [True])  # native app is built only for container
+def test_perf_comm_change(
+    system_profiler: SystemProfiler,
+    application_pid: int,
+    application_docker_container: Container,
+    perf_mode: str,
+) -> None:
+    """
+    Runs a program that changes its comm (i.e comm of the main thread).
+    The stack output by perf should use the original name, not the changed one.
+
+    This tests our modification to perf here:
+    https://github.com/Granulate/linux/commit/40a7823cf90a7e69ce8af88d224dfdd7e371de2d
+
+    perf records "comm" events - changes of task comms. Samples printed for a thread will use the current comm
+    perf understands that it has. Our modification makes it select a PERF_RECORD_COMM event that has the "exec"
+    flag. Below are 2 events from the native program - the first is the name it was executed with, then
+    the rename. If perf has started before the app, we expect to see the exec name and not the current name.
+
+        native 1901079 [010] 984807.648525: PERF_RECORD_COMM exec: native:1901079/1901079
+        oative 1901079 [010] 984807.648792: PERF_RECORD_COMM: oative:1901079/1901079
+
+    If perf started after the app, we will see the changed name (although I'd prefer to see the original name, but
+    I'm not sure it can be done, i.e is this info even kept anywhere).
+    """
+    with system_profiler as profiler:
+        collapsed = snapshot_pid_collapsed(profiler, application_pid)
+        # first run - we get the changed name, because the app started before perf began recording.
+        assert is_function_in_collapsed("oative;", collapsed)
+        assert not is_function_in_collapsed("native;", collapsed)
+
+        # now, while perf is still running & recording, we restart the app
+        application_docker_container.restart(timeout=0)
+        application_docker_container.reload()  # post restart
+        application_pid = application_docker_container.attrs["State"]["Pid"]
+
+        # second run - we get the original name, because the app started after perf began recording.
+        collapsed = snapshot_pid_collapsed(profiler, application_pid)
+        assert not is_function_in_collapsed("oative;", collapsed)
+        assert is_function_in_collapsed("native;", collapsed)
+
+
+@pytest.mark.xfail(reason="https://github.com/Granulate/gprofiler/issues/431")
+@pytest.mark.parametrize("runtime", ["native_thread_comm"])
+@pytest.mark.parametrize("perf_mode", ["fp"])  # only fp is enough
+@pytest.mark.parametrize("in_container", [True])  # native app is built only for container
+def test_perf_thread_comm_is_process_comm(
```

Comment:
Hmm this ended up being the exact same test code :dumb: only different comments and parametrization. I'll merge somehow